### PR TITLE
Add packages to LZMA for offline class library restore support

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/aspnet/Universe/issues/1123.

List of added packages:
```
netstandard.library\2.0.1
system.buffers\4.4.0
system.numerics.vectors\4.4.0
system.security.cryptography.cng\4.4.0
```

This is about 20MB difference

### CODE REVIEW

File: build/PackageArchive.targets [hunk 1]
```diff
@@ -4,48 +4,68 @@
...
-      <GeneratedFallbackRestoreSourcesPropsPath>$(_WorkRoot)restoresources.$(LzmaOutputPackageName).props</GeneratedFallbackRestoreSourcesPropsPath>
-    </PropertyGroup>
-
-    <ItemGroup>
-      <_FallbackArchiveRestoreSources Include="$(RestoreSources)" />
-      <_FallbackArchiveRestoreSources Include="$(BuildDir)" Condition="Exists($(BuildDir))" />
-      <_FallbackArchiveRestoreSources Include="$(MetapackageRestoreSource)" Condition="Exists($(MetapackageRestoreSource))" />
-    </ItemGroup>
+    <!-- Add metapackage references to project -->
+    <RepoTasks.AddMetapackageReferences
```
Comment (JunTaoLuo) [inline build/PackageArchive.targets]: Lots of redundant references but this guarantees we have everything for both netstandard2.0 and netcoreapp2.1.

File: build/PackageArchive.targets [hunk 2]
```diff
@@ -4,48 +4,68 @@
...
     <MSBuild
       Projects="$(_WorkRoot)Archive.csproj"
       Targets="Restore"
-      Properties="RestorePackagesPath=$(FallbackStagingDir);RuntimeFrameworkVersion=$(MicrosoftNETCoreApp21PackageVersion);DotNetRestoreSourcePropsPath=$(GeneratedFallbackRestoreSourcesPropsPath);DotNetBuildOffline=true;AspNetUniverseBuildOffline=true" />
+      Properties="ArchiveTargetFramework=$(ArchiveTargetFramework);RestorePackagesPath=$(FallbackStagingDir);RuntimeFrameworkVersion=$(MicrosoftNETCoreApp21PackageVersion);DotNetRestoreSourcePropsPath=$(GeneratedFallbackRestoreSourcesPropsPath);DotNetBuildOffline=true;AspNetUniverseBuildOffline=true" />
+  </Target>
+
+  <Target Name="BuildFallbackArchive" DependsOnTargets="ResolveRepoInfo">
+    <ItemGroup>
+      <_ArchiveTargetFrameworks Include="netstandard2.0;netcoreapp2.1"/>
```
Comment (JunTaoLuo) [inline build/PackageArchive.targets]: We want to be able to restore for netstandard2.0 for class libraries and netcoreapp2.1 for apps.

File: build/tools/templates/Archive/Archive.csproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$(DotNetRestoreSourcePropsPath)" Condition="'$(DotNetRestoreSourcePropsPath)' != ''"/>
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>$(ArchiveTargetFramework)</TargetFramework>
```
Comment (natemcmaster) [inline build/tools/templates/Archive/Archive.csproj]: Did you consider multi-targeting this project instead?
```
<TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>
```

Comment (JunTaoLuo) [reply build/tools/templates/Archive/Archive.csproj]: Requires conditional metapackage references but yes that would work.

**Comment by JunTaoLuo** [comment]
This is now bumped out to 2.2.0 so I'll finish this up later.

**Comment by JunTaoLuo** [comment]
Decided not to take this for now

**Pull request closed without merging**